rsconf/component/__init__.py

```diff
@@ -224,10 +226,14 @@ def install_resource(self, name, j2_ctx=None, host_path=None):
             if host_path.ext == ".sh":
                 host_path = host_path.new(ext="")
             name = self.module_name + "/" + name.basename
-        self._bash_append_and_dst(
-            host_path,
-            file_contents=self._render_resource(name, j2_ctx),
-        )
+        try:
+            j2_ctx.this = j2_ctx[self.name]
```

Comment:
I think `this` is going to be hard to find. Also this should be done in _render_file, and can be a shallow copy so we don't modify global state.

Reply:
Fix for copy and move to render_file 0784532.

Is there a name you like? `this_component`?

Reply:
Leave it as this. It'll be easy enough to find, and better than self.